guides/source/action_view_overview.md

```diff
@@ -599,22 +599,36 @@ enables compact partial-local default variable assignments:
 <% end %>
 ```
 
-INFO: By default, partials will accept any `locals` as keyword arguments. To
-define what `locals` a partial accepts, use a `locals:` magic comment. To learn
-more, read about [Strict Locals](#strict-locals).
-
 [local_assigns]:
     https://api.rubyonrails.org/classes/ActionView/Template.html#method-i-local_assigns
 
 ### Strict Locals
 
-Action View partials will accept any number of `locals` as keyword arguments.
+Action View partials are compiled into regular Ruby methods under the hood.
+Because of a Ruby limitation, every single combinaison of `locals` passed to
+a partial requires compiling another version:
```

Comment:
I don't mind this wording, but don't really see it as a Ruby limitation so much as a quirk of Ruby's syntax

```suggestion
In Ruby method calls and local variable access look the same, so every single
combination of `locals` passed to a partial requires compiling another version:
```

If you don't think that's clearer though, yours is fine. 👍